reporters/cloud-run-scripts/CloudRunnerHub.js

```diff
@@ -69,34 +68,69 @@ app.post("/", async (req, res) => {
     // Loop over all config objects in the config file and for each append a call promise to the promiseArray.
     let promiseArray = [];
     for (const botName in configObject) {
-      const botConfig = appendBlockNumberEnvVars(configObject[botName], lastQueriedBlockNumber, latestBlockNumber);
-      console.log("Appending to promise array", botName); // TODO: refine this logging with a winston log
-      // TODO: when running locally we can execute a JSON call to a local restful API. Refactor this for intergration tests.
-      // promiseArray.push(_postJson(process.env.PROTOCOL_RUNNER_URL, botConfig));
-      promiseArray.push(_executeCloudRun(process.env.PROTOCOL_RUNNER_URL, botConfig));
+      const botConfig = _appendBlockNumberEnvVars(configObject[botName], lastQueriedBlockNumber, latestBlockNumber);
+      if (process.env.ENVIRONMENT == "production") {
+        promiseArray.push(_executeCloudRun(process.env.PROTOCOL_RUNNER_URL, botConfig));
+      } else {
+        promiseArray.push(_postJson(process.env.PROTOCOL_RUNNER_URL, botConfig));
+      }
```

Comment:
Is this the statement that's called when `ENVIRONMENT === "serverless"`? I thought this was supposed to be called only when running locally so this would hit the localhost URL?

I was under the impression that there were three environments:
- production: running on a single machine in google cloud (or elsewhere?)
- serverless: running in cloud run
- local/test/development: running in a local sandbox for the purposes of testing/experimentation.

Maybe we should document/explain these somewhere? Maybe in the appendix of bot parameters?

Reply:
Ye, you are right. this is pretty confusing & could defiantly be refined. The reason it's like this is because the hub is actually running in `production` mode and the spoke is running in `serverless` mode. I defiantly can improve this definition and improve the documentation around these settings.

Reply:
Is there a reason the hub _needs_ to be running in `production` mode? I assume since we're adding the `serverless` mode it's pretty easy to tweak how it behaves... unless there's some setting in the hub that _needs_ to be different from the spokes?

Reply:
Ye, there is a _strict_ requirement that it has the mode set to `production` (or `serverless`) as without this GCP will not catch the winson logs with:
```
const { LoggingWinston } = require("@google-cloud/logging-winston");
  require("@google-cloud/trace-agent").start();
  transports.push(new LoggingWinston());
```

However, I do think there are refinements that can be made here to make this clearer. Will report back with a solution.